flow/templates/print_status.sh

```diff
@@ -0,0 +1,55 @@
+{# The following variables are available to all scripts. #}
+{% set format_progress_bar = ("%%-%ss  %%-%ss" | format(label_length,bar_length,)) %}
+{% if detailed %}
+{% set alias = scheduler_status_code %}
+{% set operation_length = [operation_length,5] | max %}
+{% set format_job_id = ("%%-%ss  " | format(id_length,)) %}
+{% set format_operation = ("%%-%ss %%3s  " | format(operation_length,)) %}
+{% set format_operation_title = ("%%-%ss  " | format(operation_length+4,)) %}
+{% set format_label = ("%%-%ss" | format(total_label_length,)) %}
+{% set ns = namespace(dash='', format_parameters='') %}
+{% if parameters %}
+{% for para in parameters %}
+{% set ns.dash = ns.dash + ("%s  " | format('-' * parameters_length[loop.index-1])) %}
+{% set ns.format_parameters = ns.format_parameters + ("%%-%ss  " | format(parameters_length[loop.index-1],)) %}
+{% endfor %}
+{% set para_head = ns.format_parameters | format(*parameters,) %}
+{% endif %}
+{% set format_head = format_job_id + format_operation_title + '%s' + format_label %}
+{% set format_row = format_job_id + format_operation + '%s' + format_label %}
+{% endif %}
+{% block body %}
+{% if overview %}
+# Overview:
+Total # of jobs: {{'%s' |format(jobs_count)}}
+
+{{ format_progress_bar | format('label','ratio')}}
+{{ format_progress_bar | format('-'*label_length,'-'*bar_length)}}
+{% for label in progress_sorted %}
+{{ format_progress_bar | format(label[0], label[1] | draw_progressbar(jobs_count))}}
+{% endfor %}
+{% endif %}
+{% if detailed %}
+
+# Detailed View:
+{{ format_head | format('job_id', 'operation', para_head, 'labels',) }}
+{{ format_head | format('-'*id_length, '-'*(operation_length+4), ns.dash, '-'*total_label_length,) }}
+{% for job in jobs %}
+{% if parameters %}
+{% set para_output = ns.format_parameters | format(*job['parameters'].values()) %}
+{% endif %}
+{% set ns.first_row = True %}
```

Comment:
Jinja has some tricks to handle this kind of stuff. For example, you can replace this variable and use `loop.first`, see here: http://jinja.pocoo.org/docs/2.10/templates/#for

Reply:
Sorry I did made the change but had to revert that later and forgot to post it here. With the current implementation we still need the ns.first_row since this is not print differently for the first operation within the loop but the first qualified one in the loop. But based on what you suggested today if we have a seperate filter logic before the table printing then we may then use loop.first.